optuna/_experimental.py

```diff
@@ -72,10 +77,12 @@ def _experimental_wrapper(f: Any) -> Any:
 
         def _experimental_func(func: Callable[[Any], Any]) -> Callable[[Any], Any]:
 
-            docstring = _EXPERIMENTAL_DOCSTRING_TEMPLATE.format(ver=version)
             if func.__doc__ is None:
                 func.__doc__ = ""
-            func.__doc__ += docstring
+
+            docstring = _EXPERIMENTAL_DOCSTRING_TEMPLATE.format(ver=version)
+            indent = _get_indent(func.__doc__)
+            func.__doc__ = func.__doc__.strip() + textwrap.indent(docstring, indent) + indent
 
             # TODO(crcrpar): Annotate this correctly.
```

Comment:
@crcrpar What does this comment mean?

Reply:
I thought `def new_func(*args: Any, **kwargs: Any) -> Any`, `return func(*args, **kwargs)  # type: ignore
`, etc. were so lazy that they should be improved in the future.